### PR TITLE
Added the docs for creating Jenkins slave & running containers - [ WIP ]

### PR DESCRIPTION
**Pull request by athiruma**
1. Connect the Jenkins slave to the Jenkins master [ assumed master already running ].
2. Create ElasticSearch, Kibana, and Grafana Containers using the podman or docker-compose

### CODE REVIEW

File: jenkins/dockerc-compose.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+version: '2.2'
+
+services:
+  grafana:
+    image: quay.io/mimehta/custom-grafana:latest
```
Comment (ebattat) [inline jenkins/dockerc-compose.yml:5]: Please take latest grafana 9.5.2 from https://hub.docker.com/r/grafana/grafana

File: jenkins/dockerc-compose.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    container_name: grafana
+    ports:
+      - "3000:3000"
+    networks:
+      - monitoring-net
+    volumes:
+      - grafana-data:/var/lib/grafana
+
+  elasticsearch:
+    image: docker.elastic.co/elasticsearch/elasticsearch:7.16.0
```
Comment (ebattat) [inline jenkins/dockerc-compose.yml:15]: Update to latest stable image

File: jenkins/run_containers_pod.yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
+apiVersion: v1
+kind: Pod
+metadata:
+  name: cloud-governance-pod
+spec:
+  containers:
+    - name: elasticsearch
+      image: docker.elastic.co/elasticsearch/elasticsearch:7.16.0
```
Comment (ebattat) [inline jenkins/run_containers_pod.yml:8]: Update to latest stable image

File: jenkins/README.md [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+docker-compose up -d
+```
+
+#### Using the podman 
+
+create elasticsearch & grafana local persistence Directories
+
+Create and Allow Permissions
+```commandline
+mkdir -p /tmp/grafana
```
Comment (ebattat) [inline jenkins/README.md:92]: Data should saved on:
`/cloud_governance_data/ and not under /tmp`

Please fix it for all

File: jenkins/README.md [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+mkdir -p /tmp/elasticsearch
```
Comment (ebattat) [inline jenkins/README.md:93]: Update it as following:

CLOUD_GOVERNANCE_PATH=""

mkdir -p "$CLOUD_GOVERNANCE_PATH"/elastic/
mkdir -p "$CLOUD_GOVERNANCE_PATH"/grafana/

# chmod solved ci_pod degraded
chmod 775 -R "$CLOUD_GOVERNANCE_PATH"/elastic/
chmod 775 -R "$CLOUD_GOVERNANCE_PATH"/grafana/

For more details [see](https://github.com/redhat-performance/benchmark-runner/blob/ecdb4396dd1b1f708ff1a5d794f043aa8d449c63/ci_pod/create_ci_pod.sh)

**Review by ebattat — COMMENTED** [review]
Please see my comments

File: jenkins/run_containers_pod.yml [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+    - name: kibana
+      image: docker.elastic.co/kibana/kibana:8.8.0
+      ports:
+        - containerPort: 5601
+          hostPort: 5601
+      env:
+        - name: ELASTICSEARCH_HOSTS
+          value: http://localhost:9200
+    - name: grafana
+      image: grafana/grafana
```
Comment (ebattat) [inline jenkins/run_containers_pod.yml:29]: Take a specific version, not latest

File: jenkins/run_containers_pod.yml [hunk 7]
```diff
@@ -0,0 +1,46 @@
+apiVersion: v1
+kind: Pod
+metadata:
+  name: cloud-governance-pod-1
```
Comment (ebattat) [inline jenkins/run_containers_pod.yml:4]: cloud-governance-pod-1 - why we need -1 ?

File: jenkins/dockerc-compose.yml [hunk 8]
```diff
@@ -0,0 +1,40 @@
+version: '2.2'
+
+services:
+  grafana:
+    image: grafana/grafana
```
Comment (ebattat) [inline jenkins/dockerc-compose.yml:5]: Take a specific version, not latest

File: jenkins/run_containers_pod.yml (hunk 7 above)
Comment (athiruma) [reply jenkins/run_containers_pod.yml]: Ohh, sorry. Forgot to delete 1 after testing

File: jenkins/Step_by_Step.md [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+2. Install docker on [Fedora](https://docs.docker.com/engine/install/fedora/)
+   ```commandline
+    sudo dnf -y install dnf-plugins-core
+    sudo dnf config-manager --add-repo https://download.docker.com/linux/fedora/docker-ce.repo
+    sudo dnf install docker-ce docker-ce-cli containerd.io docker-buildx-plugin docker-compose-plugin docker-compose
+    sudo systemctl start docker   
+    ```
+3. Create Jenkins user
+   ```commandline
+    useradd jenkins -U -s /bin/bash
```
Comment (ebattat) [inline jenkins/Step_by_Step.md:21]: extra space

File: jenkins/Step_by_Step.md [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+   ```
+5. Giving permissions to jenkins user to run docker container
+    ```
+   sudo chown jenkins:jenkins /var/run/docker.sock
+    ```
+
+6. Run the cloud_governance_stack [ ElasticSearch, Kibana, Grafana]
+    ```commandline
+    # detached mode
+    docker-compose -f docker_compose_file_path up -d
```
Comment (ebattat) [inline jenkins/Step_by_Step.md:37]: pls put the link to d[ocker-compose.yml](https://github.com/redhat-performance/cloud-governance/pull/602/files#diff-0b3bf01515c785027e95f6dcb6cbec90e305189dc3c25a0c522d4070eaa743a4)

File: jenkins/Step_by_Step.md [hunk 11]
```diff
@@ -0,0 +1,88 @@
...
+4. Select Manage Credentials
+5. Click on **System**, select the domain that your creds will be stored
+   1. Add Credentials.
+      1. Select **secret file**
+      2. Give the Id
+      3. Upload the json file
+   2. Update Credentials
+      1. Select the secret you want to upgrade.
+      2. If it is a file secret.
+      3. Upload the modified file.
```
Comment (ebattat) [inline jenkins/Step_by_Step.md:88]: What about add account for running policies ? (enable/disable policies)
What about add account for running tagging ?

File: jenkins/Step_by_Step.md [hunk 12]
```diff
@@ -0,0 +1,88 @@
...
+   sudo chown jenkins:jenkins /var/run/docker.sock
+    ```
+
+6. Run the cloud_governance_stack [ ElasticSearch, Kibana, Grafana]
+    ```commandline
+    # detached mode
+    docker-compose -f docker_compose_file_path up -d
+    # down the containers
+    docker-compose -f docker_compose_file_path down
+    ```
```
Comment (ebattat) [inline jenkins/Step_by_Step.md:40]: What about run_comtainer_pod.yml ?

Comment (athiruma) [reply jenkins/Step_by_Step.md:40]: Related to podman.
Here we are using docker.

File: jenkins/Step_by_Step.md (hunk 11 above)
Comment (athiruma) [reply jenkins/Step_by_Step.md:88]: We will use only one account for both tagging/ policies.

File: jenkins/Step_by_Step.md (hunk 12 above)
Comment (ebattat) [reply jenkins/Step_by_Step.md:40]: Maybe change the name to podman_pod.yml.
and add comment that we are using docker-compose.yml

File: jenkins/Step_by_Step.md (hunk 11 above)
Comment (ebattat) [reply jenkins/Step_by_Step.md:88]: We should give them instruction how to add more account, exact place in the code that they need to add the account.

Comment (athiruma) [reply jenkins/Step_by_Step.md:88]: Yes, I gave the instructions at the top.

**Review by ebattat — APPROVED** [review]
/LGTM

**Pull request merged**